tests/integration/workflows/test_arxiv_workflow.py

```diff
@@ -982,3 +984,102 @@ def test_start_wf_with_no_source_data_fails(workflow_app):
 
     with pytest.raises(ValueError):
         start('article', object_id=obj.id)
+
+
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.download_file_to_workflow',
+    side_effect=fake_download_file,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.actions.download_file_to_workflow',
+    side_effect=fake_download_file,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.is_pdf_link',
+    return_value=True
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.beard.json_api_request',
+    side_effect=fake_beard_api_request,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.magpie.json_api_request',
+    side_effect=fake_magpie_api_request,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.refextract.extract_references_from_file',
+    return_value=[],
+)
+def test_do_not_repeat(
+    mocked_refextract_extract_refs,
+    mocked_api_request_magpie,
+    mocked_api_request_beard,
+    mocked_is_pdf_link,
+    mocked_package_download,
+    mocked_arxiv_download,
+    workflow_app,
+    mocked_external_services,
+):
+    def return_value(val):
+        def _return_value(obj, eng):
+            return {'id': val}
+        return _return_value
+
+    custom_wf_steps = [
```

Comment:
I would add here the step that is used in the real workflows to restore from `source_data`, so you're testing the interactions between the two.